routes/registration.js

```diff
@@ -15,61 +15,105 @@ const validateEmail = utilityFunctions.validateEmail;
+            brandTitle1: mailHeaderData.brandTitle1,
+            brandTitle2: mailHeaderData.brandTitle2,
+            greeting: mailData.registration.greeting,
+            textContent1: mailData.registration.textContent.textContent1,
+            textContent2: mailData.registration.textContent.textContent2,
+            textContent3: mailData.registration.textContent.textContent3,
+            textContent4: mailData.registration.textContent.textContent4,
+            callToAction: mailData.registration.callToAction,
+            signatureGreet: mailData.registration.signature.endGreet,
+            signatureSignOff: mailData.registration.signature.signOff,
+            footerText1: mailFooterData.textContent1,
+            footerText2: mailFooterData.copyrights
+        };
+
+        const htmlToSend = template(replacements);
+        mail.mailOptions.to = req.body.email;
+        mail.mailOptions.subject = 'PC PrepKit Email Verification';
+        mail.mailOptions.html = htmlToSend;
+        mail.smtpTransport.sendMail(mail.mailOptions, function(error) {
+            if (error) {
+                res.status(500).json({error: errorCode.PCE001.message, code: errorCode.PCE001.code});
+            } else {
+                res.json(successCode.PCS005.message);
+            }
+        });
     });
 }
 // Receiving HTTP Post
 router.post('/', function(req, res) {
     if (!req.body.email || !validateEmail(req.body.email)) {
-        return res.status(400).json({error: 'Email is invalid'});
+        return res.status(400).json({error: errorCode.PCE002.message, code: errorCode.PCE002.code});
+    } else if (!req.body.fname || !validateName(req.body.fname)) {
+        return res.status(400).json({error: errorCode.PCE003.message, code: errorCode.PCE003.code});
+    } else if (!req.body.lname || !validateName(req.body.lname)) {
+        return res.status(400).json({error: errorCode.PCE004.message, code: errorCode.PCE004.code});
+    } else if (!req.body.password || !validatePassword(req.body.password)) {
+        return res.status(400).json({error: errorCode.PCE005.message, code: errorCode.PCE005.code});
+    } else {
+        verifier.verify(req.body.email, function(err, info) {
+            if (err) {
+                return res.status(400).json({error: errorCode.PCE002.message, code: errorCode.PCE002.code});
+            } else if (!info.success) {
+                return res.status(400).json({error: errorCode.PCE002.message, code: errorCode.PCE002.code});
+            } else {
+                const rString = randomStr(50, '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ');
+                localUser.create({
+                    fname: req.body.fname,
+                    lname: req.body.lname,
+                    email: req.body.email,
+                    password: req.body.password,
+                    provider: 1
+                }).catch(error => {
+                      if (error) {
```

Comment:
what happens in the case `error` is false/undefined? what would you return?

or, does it make sense to check for `error` when it is in the catch clause?

Reply:
If error is undefined, the procedure goes on without returning and fills data in other tables. If there is error in any one step it stops and sends error message. If all go successfully, it return only one success message.

Reply:
but you have a `catch` clause here with `if (error)`
is this `if` statement necessary? will `catch` clause ever be trigger if the `error` is undefined/dne?

Reply:
Okay, yes got it. Will correct it.